scoped-transitions.md

```diff
@@ -0,0 +1,110 @@
+# Scoped Transitions
+
+This doc is a summary of early design explorations to scope transitions to a DOM sub-tree. The following is a rough API sketch for how a developer would trigger the transition rooted on an element (other than the document's root element).
+
+```js
+element.createTransition({
+  updateDOM() {
+    /* … */
+  },
+});
+```
+
+The element becomes the **scoped-transition-root** for the transition. It's the element that will host the pseudo-element tree and page-transition-container sub-trees will be created for descendants that have a `page-transition-tag`. For example, the following page:
+
+```html
+<style>
+  .outer {
+    page-transition-tag: outer;
+    contain: layout;
+  }
+  .inner {
+    page-transition-tag: inner;
+    contain: layout;
+  }
+</style>
+<div class="outer">
+  <div class="inner"></div>
+</div>
+<script>
+  document.querySelector('.outer').createTransition(…);
+</script>
+```
+
+produces the following DOM tree under `outer`:
+
+```
+outer
+└─ ::page-transition
+   ├─ ::page-transition-container(outer)
+   │  └─ ::page-transition-image-wrapper(outer)
+   │     ├─ ::page-transition-outgoing-image(outer)
+   │     └─ ::page-transition-incoming-image(outer)
+   └─ ::page-transition-container(inner)
+      └─ ::page-transition-image-wrapper(inner)
+         ├─ ::page-transition-outgoing-image(inner)
+         └─ ::page-transition-incoming-image(inner)
+
+Note: We will probably need to change the naming, as 'page-transition' doesn't work.
+Note: We might want to add an ID for a transition to control which transition the `page-transition-tag` applies to.
```

Comment:
We don't do this for root transitions. Feels like we should do it for both if we think it's worth it.

Reply:
I think this is more a multiple independent transitions problem, but it's definitely something we need to figure out. 

Specifically, there is a problem with nested transitions where we have an element transition and, say, a document transition. Currently there is no way to distinguish what `page-transition-tag` means: should we tag that in the element transition or in the root transition or both? I don't think it's as easy as saying `::page-transition(transition-tag)`. 

It's possible that the page-transition-tag syntax should change to something like `page-transition-tag: (tag, [transition-tag1, transition-tag2])` or some tuple to indicate that this element has a tag that references one or more `transition-tag`s

Reply:
In a DOM structure like:

```html
<root>
  <component-a>
    <component-b></component-b>
  </component-a>
</root>
```

Why are we creating special syntax to differentiate a root transition and a scoped transition, but not worried about differentiating between transitions scoped to component-a vs component-b?

Reply:
> but not worried about differentiating between transitions scoped to component-a vs component-b

Differentiating between these is also a goal. For the example above:

```html
<root>
  <component-a>
    <div class="header-a"></div>
    <component-b>
      <div class="header-b"></div>
    </component-b>
  </component-a>
</root>
```

The syntax could be:
```css
.component-a {
  page-transition-root: a;
}
.component-b {
  page-transition-root: b;
}

.header-a {
  page-transition-tag: header a;
}

.header-b {
  page-transition-tag: header b;
}
```

Devs could always create prefixes themselves if they wanted this, `page-transition: header-a`, so its just syntactic sugar. We can ignore it for now.

Reply:
Moved to issue #195.